ambari/src/main/java/io/brooklyn/ambari/AmbariClusterImpl.java

```diff
@@ -158,12 +137,67 @@ public void installServices() {
         List<String> services = getConfig(HADOOP_SERVICES);
         List<String> hosts = getAttribute(AMBARI_SERVER).getAttribute(AmbariServer.REGISTERED_HOSTS);
 
-        LOG.debug("About to create cluster with services: " + services);
-        if (services != null && services.size() > 0) {
+        if (isServicesBasedDeployment(services)) {
+            LOG.debug("About to create cluster with services: " + services);
             getAttribute(AMBARI_SERVER).installHDP("Cluster1", "mybp", hosts, services);
         } else {
-            getAttribute(AMBARI_SERVER).installHDP("Cluster1", "mybp", hosts, DEFAULT_SERVICES);
+            AmbariConfig config = ambariConfigFromHostgroups();
+            if (config.hasHostGroups()) {
+                getAttribute(AMBARI_SERVER).installHDPFromConfig("Cluster1", "mybp", config);
```

Comment:
Eventually want the first two arguments to `installHDP` and `installHDPFromConfig` to be customisable.